### PR TITLE
Support sig-proxy for podman-remote attach and start

### PR DESCRIPTION
**Pull request by ashley-cui**
Signals were not proxied for attach and start for podman-remote. Now they are.

Signed-off-by: Ashley Cui <acui@redhat.com>

Fixes: #16662

#### Does this PR introduce a user-facing change?

```release-note
Sig-proxy is now supported for podman-remote attach and start
```

### CODE REVIEW

**Comment by ashley-cui** [comment]
@rhatdan PTAL :)

File: test/system/032-sig-proxy.bats [hunk 1]
```diff
@@ -40,4 +40,99 @@ load helpers
     run_podman rm -f -t0 foo
 }
 
+@test "podman start sigkill" {
+
+    $PODMAN create --name foo $IMAGE sh -c 'trap "echo BYE;exit 0" INT;echo READY;while :;do sleep 0.1;done'
```
Comment (edsantiago) [inline test/system/032-sig-proxy.bats:45]: Please don't use `$PODMAN` unless it's absolutely an emergency. (The next line, 46, might count as such. I'll look into it closer tomorrow. This line, 45, should use `run_podman`.

Comment (edsantiago) [reply test/system/032-sig-proxy.bats:45]: Never mind, I see what happened. It's copypasta nastiness. Fix in progress, I will report back later this morning.

Comment (edsantiago) [reply test/system/032-sig-proxy.bats:45]: @ashley-cui try [this 032-sigproxy.bats](https://github.com/containers/podman/files/10344802/032-sigproxy.bats.txt). I've confirmed that it passes (including remote) on your branch, and the two new tests fail on `main`.

Suggestion: I encourage you to form a new habit. Any time you find yourself copypasting more than a handful of lines, or if you see someone doing so, take a step back to evaluate what common elements you can refactor. Over time you will find your brain screaming at these aberrations, and you'll take time to fix them, and you will become a much better programmer. (You already are, but we all have room for improvement!)

Comment (ashley-cui) [reply test/system/032-sig-proxy.bats:45]: Thanks so much Ed!

**Comment by rhatdan** [comment]
LGTM
@containers/podman-maintainers PTAL

**Pull request merged**